### DISCUSSION TITLE
Use the same variable in front matter for thumbnail and featured image

### DISCUSSION THREAD

**Discussion by LITUATUI** [chipzoller/hugo-clarity#273] [Q&A] [answered]
Hi.

I'm in the process of migrating a WordPress blog to Hugo.

The most popular [plugin](https://github.com/SchumacherFM/wordpress-to-hugo-exporter) to convert WordPress posts to markdown generates the following variables in the front matter:

title:
author:
type:
date:
url:
featureImage:
categories:
tags:

How can I use `featureImage` variable to also be used as the source of the thumbnail? 

I really would like to avoid creating a thumbnail variable in the front matter for hundreds of posts, since I want to use the same image as the featured. 

How can I use the same variable for thumbnails and featured images? 

For example, in the [Mainroad theme](https://github.com/Vimux/Mainroad), If I set a thumbnail: image.jpg in the front matter, it's used as a thumbnail shown at the homepage and on top of a post/page.

**Comment by timschps** [comment]
Stupid question, but isn’t it an option to do a search and replace so that your requirements are met? Having the 2 variables for future use is really interesting (and just setting up 2 variables in your existing posts that each time post to the same image)

Tim
________________________________
Van: Pedro Lima ***@***.***>
Verzonden: Tuesday, March 22, 2022 9:23:19 PM
Aan: chipzoller/hugo-clarity ***@***.***>
CC: Subscribed ***@***.***>
Onderwerp: [chipzoller/hugo-clarity] Use the same variable in front matter for thumbnail and featured image (Discussion #273)


Hi.

I'm in the process of migrating a WordPress blog to Hugo.

The most popular plugin<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FSchumacherFM%2Fwordpress-to-hugo-exporter&data=04%7C01%7C%7C39e8f95d4c4f4b0058d208da0c41ce4a%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637835774020202246%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Ut1402QgwF0pWG3%2FeZkQ4Z55A7KrByGk%2FNb5WVrZPOU%3D&reserved=0> to convert WordPress posts to markdown generates the following variables in the front matter:

title:
author:
type:
date:
url:
featureImage:
categories:
tags:

How can I use featureImage variable to also be used as the source of the thumbnail?

I really would like to avoid creating a thumbnail variable in the front matter for hundreds of posts, since I want to use the same image as the featured.

How can I use the same variable for thumbnails and featured images?

For example, in the Mainroad theme<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FVimux%2FMainroad&data=04%7C01%7C%7C39e8f95d4c4f4b0058d208da0c41ce4a%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637835774020202246%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=JpS%2Fxp1WScLQe31DUA2dSsIQ%2BzFusIvBYd%2Fmq9c0gF8%3D&reserved=0>, If I set a thumbnail: image.jpg in the front matter, it's used as a thumbnail shown at the homepage and on top of a post/page.

**Reply by LITUATUI** [reply]
I can do a batch search and replace a variable with VS Code, I've done that to try multiple themes that use different variable names for featured images. The problem is that I only have one variable name available in the front matter of hundreds of markdown files. 

If I need to use an extra variable I have to create it manually. It wouldn't be a problem if it was a new blog.

**Comment by timschps** [comment, marked as answer]
You can use the replace functionality in vs code with regex to replace a one line variable to a 2 variable-2line result. You “just” need to tweak your search 😉
________________________________
Van: Pedro Lima ***@***.***>
Verzonden: Tuesday, March 22, 2022 9:47:01 PM
Aan: chipzoller/hugo-clarity ***@***.***>
CC: Tim Schaeps ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [chipzoller/hugo-clarity] Use the same variable in front matter for thumbnail and featured image (Discussion #273)


I can do a batch search and replace a variable with VS Code, I've done that to try multiple themes that use different variable names for featured images. The problem is that I only have one variable name available in the front matter of hundreds of markdown files.

If I need to use an extra variable I have to create it manually. It wouldn't be a problem if it was a new blog.

**Reply by LITUATUI** [reply]
Thanks for the suggestion, I'll try that.
